frontend/src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service.ts

```diff
@@ -5,76 +5,131 @@
 
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { BehaviorSubject, Observable, tap } from 'rxjs';
-import { ToastService } from 'src/app/helpers/toast/toast.service';
+import {
+  AbstractControl,
+  AsyncValidatorFn,
+  ValidationErrors,
+} from '@angular/forms';
+import { BehaviorSubject, Observable, map, take, tap } from 'rxjs';
 import { T4CInstance } from 'src/app/services/settings/t4c-instance.service';
 import { environment } from 'src/environments/environment';
 
 @Injectable({
   providedIn: 'root',
 })
 export class T4CRepoService {
-  constructor(private http: HttpClient, private toastService: ToastService) {}
+  constructor(private http: HttpClient) {}
 
-  _repositories = new BehaviorSubject<T4CServerRepository[]>([]);
-  get repositories(): T4CServerRepository[] {
-    return this._repositories.value;
+  private _repositories = new BehaviorSubject<
+    T4CServerRepository[] | undefined
+  >(undefined);
+
+  readonly repositories = this._repositories.asObservable();
+
+  instanceUrlFactory(instanceId: number): string {
+    return `${environment.backend_url}/settings/modelsources/t4c/${instanceId}/repositories/`;
```

Comment:
```suggestion
    return `${environment.backend_url}/settings/modelsources/t4c/${instanceId}/repositories`;
```

Don't use a trailing `/` in the urlFactory. Two reasons: 
- When just using the urlFactory as URL in requests, the request is done with a trailing slash. This leads to a redirect in fastAPI and therefore another (unnecessary) request. 
- Concatenation looks weird, e.g., `${this.repositoryUrlFactory(instanceId, repositoryId)}start/` instead of  `${this.repositoryUrlFactory(instanceId, repositoryId)}/start` (the latter looks a lot better imho)